doc/adr/0001-messages-to-find-case-law.md

```diff
@@ -0,0 +1,46 @@
+# 1. Messages to Find Case Law
+
+Date: 2023-03-01
+
+## Status
+
+Very Draft
+
+## Context
+
+Find Case Law (FCL) would like to be able to send messages to the Transformation Engine (TRE)
+to request parsing of judgments (either ones previously sent by Transfer Digital Records (TDR) or
+novel ones), and TRE to respond with a parsed judgment and associated metadata from the TRE processes.
+
+TRE would like to have proper schemas for the information that flows between TRE and FCL, and not
+store/receive irrelevant information.
+
+## Decision
+
+There is a new format for the messages sent via SQS/SNS -- see commit "Discussed changes to the proposed messages"
+
+Whilst AVRO is being used to define the schema, the actual messages are likely to be JSON.
+
+When FCL would like a judgment in DOCX format parsing, it will send a link to a DOCX file
+(which will probably be a signed S3 link, but this should be below the abstraction layer)
+along with an opaque reference identifier controlled by FCL which will be sent back.
+No other metadata (for example, FCLs record of the original submitter of the record to TDR) will be sent,
+as TRE does not want to have to handle it. The identifier will be passed back by TRE to FCL,
+and will also note the originator of the file was Find Case Law.
+
+The [schema](../../tre_schemas/avro/fcl-judgment-parse-request.avsc) and [example json](../../json-examples/fcl-judgment-parse-request.json)
+for this look good.
+
+In response to either this request or a new judgment from a clerk submitted to TDR, a message will be
+sent [schema](../../tre_schemas/avro/tre-judgment-available.avsc), [example json](../../json-examples/tre-judgment-available.json).
+
+The response will continue to point to a tar.gz file with metadata in the previous format; however, FCL MUST support
+the TDR section of the metadata.json being absent for documents that don't come from FCL.
+
+TRE will maintain original format messages for the forseeable future.
+
+## Consequences
+
+In the current absence of a permanent accessible store of received judgments, FCL will need to apply
```

Comment:
When we ingest the data from TRE, we won't have values that would have come from TDR:

Source-Organisation
Contact-Name
Contact-Email
Internal-Sender-Identifier
Consignment-Completed-Datetime

which either:
- for existing judgments we already have in MarkLogic and should preserve, or
- for new-to-us judgments we might need to inject either blank values or actual values if known

Ideally I'd like the re-parser emitter (which sends the docx files to TRE) to send these values to TRE which would reflect them, so that they can be configured at runtime (so we can retain the provinance information), but that feels non-ideal for TRE; we will probably need to add those values (empty or otherwise) in the ingester.